### ISSUE TITLE
node-oracledb 6.0.3 is now available!

### ISSUE THREAD

**Issue by sharadraju** [oracle/node-oracledb#1587]
node-oracledb v6.0.3 is out!
See the [CHANGELOG](https://node-oracledb.readthedocs.io/en/latest/release_notes.html) for details.
This release contains bug fixes for the previous 6.0.x releases, including the patches provided for the issues raised by our users. Thank you for reporting the issues promptly!

**Comment by hinogi** [comment]
Is there a migration guide what changes would be necessary to change from 5.x to 6.x? I saw some reports on failing db connections when people tried to use it with typeorm for example.

**Comment by cjbj** [comment]
@hinogi start with the documentation [Upgrading from node-oracledb 5.5 to 6.0](https://node-oracledb.readthedocs.io/en/latest/user_guide/migrate.html#upgrading-from-node-oracledb-5-5-to-6-0) and see how you go.

**Comment by hinogi** [comment]
can someone maybe help with typeorm support for oracle 6? I guess the default was thick mode before so currently this won't work with v6.

**Comment by sharadraju** [comment]
> I guess the default was thick mode before so currently this won't work with v6.

There was no Thick mode or Thin mode before node-oracledb v6.

To enable thick mode, you simply have to have a call to `initOracleClient()` before making a connection. See https://node-oracledb.readthedocs.io/en/latest/user_guide/initialization.html#enabling-node-oracledb-thick-mode.
By default the basic tests of TypeORM should run in the default 'Thin' mode of v6 and beyond. 

> I saw some reports on failing db connections when people tried to use it with typeorm for example.

I guess you are talking about the latest comments from Issue #1565 on TypeORM with a TypeORM user. we are currently loking into it. In the meantime, you can keep using the Thick mode.

**Comment by hinogi** [comment]
okay, let me rephrase, before the db connection relied on the instant client. Currently, user who may still use an older db version before 12.1 will run into problems.
TypeORM for example just does ` return require("oracledb")` to load the driver that will not work anymore.
Maybe you could add some info in here https://github.com/typeorm/typeorm/issues/10277

**Comment by cjbj** [comment]
@hinogi what problems with Thin?  The issue in #1565 looks like a test issue due to the schema being recreated.

**Comment by hinogi** [comment]
As far as I can see it, there are a mix of error. Maybe especially for those with older db versions. The default before was with instant client and not db version restrictions, so drop-in replacement does not work. and people who tried run into errors.

**Comment by hinogi** [comment]
Is there a way to make a version check and switch to thick mode if the version is not 12.1 < ?

**Comment by hinogi** [comment]
That sound doable :)

**Comment by anthony-tuininga** [comment]
Just to clarify, thin mode does not support connecting to Oracle 11g so checking `oracleServerVersionString` will not actually work. Instead you need to catch the exception `NJS-138: connections to this database server version are not supported by node-oracledb in Thin mode` and then make the attempt to enable thick mode. In other words, something like this:

```javascript
async function connect(config) {
  try {
    return await oracledb.getConnection(config);
  } catch (err) {
    if (err.code !== 'NJS-138' || !oracledb.thin)
     throw(err);
  }
  oracledb.initOracleClient();
  return await oracledb.getConnection(config);
}
```
